src/category_theory/subobject.lean

```diff
@@ -918,13 +1317,38 @@ instance {B : C} : semilattice_sup (subobject B) :=
   sup_le := λ m n k, quotient.induction_on₃' m n k (λ a b c ⟨i⟩ ⟨j⟩, ⟨mono_over.sup_le _ _ _ i j⟩),
   ..subobject.partial_order B }
 
+lemma sup_factors_of_factors_left {A B : C} {X Y : subobject B} {f : A ⟶ B} (P : X.factors f) :
+  (X ⊔ Y).factors f :=
+(subobject.factors_iff _ _).mpr
+⟨factor_thru _ _ P ≫ underlying.map (hom_of_le le_sup_left), by simp [factor_thru_arrow _ _ P]⟩
+
+lemma sup_factors_of_factors_right {A B : C} {X Y : subobject B} {f : A ⟶ B} (P : Y.factors f) :
+  (X ⊔ Y).factors f :=
+(subobject.factors_iff _ _).mpr
+⟨factor_thru _ _ P ≫ underlying.map (hom_of_le le_sup_right), by simp [factor_thru_arrow _ _ P]⟩
+
 section
 variables [has_zero_morphisms C] [has_zero_object C]
 
 instance {B : C} : semilattice_sup_bot (subobject B) :=
 { ..subobject.order_bot,
   ..subobject.semilattice_sup }
 
+lemma finset_sup_factors {I : Type*} {A B : C} {s : finset I} {P : I → subobject B}
```

Comment:
We talked before about having two instances of `semilattice_sup_bot`, which seems reasonable, but I'm unsure about having two versions of this lemma; one assuming [has_zero_morphisms C] and the other assuming cartesian closed-ness. Do you think we can do something to avoid this duplication?

Reply:
I've added a comment, punting on this.